rpcclient/wallet.go

```diff
@@ -1279,6 +1279,96 @@ func (c *Client) CreateMultisig(requiredSigs int, addresses []dcrutil.Address) (
 	return c.CreateMultisigAsync(requiredSigs, addresses).Receive()
 }
 
+// FutureCreateUnsignedTicketResult is a future promise to deliver the result of a
+// CreateUnsignedTicket RPC invocation (or an applicable error).
+type FutureCreateUnsignedTicketResult chan *response
+
+// Receive waits for the response promised by the future and returns the
+// result of creating unsigned ticket.
+func (r FutureCreateUnsignedTicketResult) Receive() ([]string, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmashal result as a string slice.
+	var unsignedTickets []string
+	err = json.Unmarshal(res, &unsignedTickets)
+	if err != nil {
+		return nil, err
+	}
+
+	return unsignedTickets, nil
+}
+
+// CreateUnsignedTicketAsync returns an instance of a type that can be used to get the
+// result of the RPC at some future time by invoking the Receive function on the
+// returned instance.
+//
+// See CreateUnsignedTicket for the blocking version and more details.
+func (c *Client) CreateUnsignedTicketAsync(fromAccount string,
+	spendLimit dcrutil.Amount, minConf *int, ticketAddress dcrutil.Address,
+	numTickets *int, poolAddress dcrutil.Address, poolFees *dcrutil.Amount,
+	expiry *int, ticketFee *dcrutil.Amount) FutureCreateUnsignedTicketResult {
+
+	// An empty string is used to keep the sendCmd
+	// passing of the command from accidentally
+	// removing certain fields. We fill in the
+	// default values of the other arguments as
+	// well for the same reason.
+
+	minConfVal := 1
+	if minConf != nil {
+		minConfVal = *minConf
+	}
+
+	ticketAddrStr := ""
+	if ticketAddress != nil {
+		ticketAddrStr = ticketAddress.EncodeAddress()
+	}
+
+	numTicketsVal := 1
+	if numTickets != nil {
+		numTicketsVal = *numTickets
+	}
+
+	poolAddrStr := ""
+	if poolAddress != nil {
+		poolAddrStr = poolAddress.EncodeAddress()
+	}
+
+	poolFeesFloat := 0.0
+	if poolFees != nil {
+		poolFeesFloat = poolFees.ToCoin()
+	}
+
+	expiryVal := 0
+	if expiry != nil {
+		expiryVal = *expiry
+	}
+
+	ticketFeeFloat := 0.0
+	if ticketFee != nil {
+		ticketFeeFloat = ticketFee.ToCoin()
+	}
+
+	cmd := dcrjson.NewCreateUnsignedTicketCmd(fromAccount, spendLimit.ToCoin(),
+		&minConfVal, &ticketAddrStr, &numTicketsVal, &poolAddrStr,
+		&poolFeesFloat, &expiryVal, dcrjson.String(""), &ticketFeeFloat)
+
+	return c.sendCmd(cmd)
+}
+
+// CreateNewAccount creates a new wallet account.
+func (c *Client) CreateUnsignedTicket(fromAccount string,
```

Comment:
I think `CreateUnsignedTicket` should also return `[]*chainhash.Hash` for consistency, since it's identical to `PurchaseTicket`, sans signing.

Reply:
You think it should return `[]*chainhash.Hash` and the encoded unsigned transaction?

Reply:
oops, disregard that. Returning the encoded unsigned transaction(s) should do it.